database/migrations/2017_07_31_153255_add_campaign_id_to_posts.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+use Rogue\Models\Post;
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class AddCampaignIdToPosts extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('posts', function (Blueprint $table) {
+            $table->integer('campaign_id')->after('signup_id')->index()
+                ->comment('The campaign node ID for this post.');
+        });
+
+        // Copy `campaign_id` from the signups table so we can make more
+        // efficient queries against the posts table (without expensive join).
+        foreach (Post::with('signup')->cursor() as $post) {
```

Comment:
I think this belongs in a migration, since it lets us ensure that the schema change happens alongside the corresponding logic changes. And unless there's a misconfiguration in Jenkins, it shouldn't be possible to run more than one deploy at once.